starlette/applications.py

```diff
@@ -10,6 +10,20 @@
 from starlette.routing import BaseRoute, Router
 from starlette.types import ASGIApp, Receive, Scope, Send
 
+_ExcKey = typing.TypeVar(
+    "_ExcKey",
+    bound=typing.Union[int, typing.Type[Exception]],
+    contravariant=True,
+)
+# Breaking out Exception to a bounded TypeVar here allows defining a narrower type in
+# handlers, like HTTPException.
+_HandledException = typing.TypeVar("_HandledException", bound=Exception)
+_ExcHandler = typing.Callable[
+    [Request, _HandledException], typing.Union[Response, typing.Awaitable[Response]]
+]
+_ExcDict = typing.Dict[_ExcKey, _ExcHandler]
```

Comment:
Could we get rid of this type alias and just be more verbose in the signatures / do something like `self.exception_handlers = dict(exception_handlers or {})` which I think mypy will understand?